tests/test_packages/test_connections/test_p2p_libp2p/test_integration.py

```diff
@@ -42,204 +39,114 @@
 
 DEFAULT_NET_SIZE = 4
 
-MockDefaultMessageProtocol = Mock()
-MockDefaultMessageProtocol.protocol_id = DefaultMessage.protocol_id
-MockDefaultMessageProtocol.protocol_specification_id = (
-    DefaultMessage.protocol_specification_id
-)
-
 
 @pytest.mark.integration
 @libp2p_log_on_failure_all
 class TestP2PLibp2pConnectionIntegrationTest(BaseP2PLibp2pTest):
     """Test mix of relay/delegate agents and client connections work together"""
 
     @classmethod
-    def setup_class(cls):
-        """Set the test up"""
-        super().setup_class()
-
-        cls.multiplexers_dict = {}
-        cls.connections_dict = {}
-
-        try:
-            cls.main_relay = cls.make_connection("main_relay", relay=True)
-            main_relay = cls.main_relay.node.multiaddrs[0]
-            cls.relay_2 = cls.make_connection(
-                "relay_2", entry_peers=[main_relay], relay=True
-            )
-            relay_peer_2 = cls.relay_2.node.multiaddrs[0]
-
-            cls.delegate_1 = cls.make_connection(
-                "delegate_1",
-                entry_peers=[main_relay],
-                relay=True,
-                delegate=True,
-                mailbox=True,
-            )
-
-            cls.delegate_2 = cls.make_connection(
-                "delegate_2",
-                entry_peers=[relay_peer_2],
-                relay=True,
-                delegate=True,
-                mailbox=True,
-            )
-
-            cls.agent_connection_1 = cls.make_connection(
-                "agent_connection_1",
-                entry_peers=[main_relay],
-                relay=False,
-                delegate=False,
-            )
-
-            cls.agent_connection_2 = cls.make_connection(
-                "agent_connection_2",
-                entry_peers=[relay_peer_2],
-                relay=False,
-                delegate=False,
-            )
-            cls.client_connection_1 = cls.make_client_connection(
-                "client_1",
-                peer_public_key=cls.delegate_1.node.pub,
-                **cls.get_delegate_host_port(cls.delegate_1.node.delegate_uri),
-            )
-
-            cls.client_connection_2 = cls.make_client_connection(
-                "client_2",
-                peer_public_key=cls.delegate_2.node.pub,
-                **cls.get_delegate_host_port(cls.delegate_2.node.delegate_uri),
-            )
-            cls.mailbox_connection_1 = cls.make_mailbox_connection(
-                "mailbox_1",
-                peer_public_key=cls.delegate_1.node.pub,
-                **cls.get_delegate_host_port(Uri(cls.delegate_1.node.mailbox_uri)),
-            )
-
-            cls.mailbox_connection_2 = cls.make_mailbox_connection(
-                "mailbox_2",
-                peer_public_key=cls.delegate_2.node.pub,
-                **cls.get_delegate_host_port(Uri(cls.delegate_2.node.mailbox_uri)),
-            )
-        except Exception:
-            cls.teardown_class()
-            raise
```

Comment:
all different variations of try / except / finally have been removed. We use `atexit` to register teardown_class which ensures that any and all multiplexers and connections disconnect, no matter how the process or python interpreter is terminated.